pages/[user].tsx

```diff
@@ -89,51 +105,36 @@ export default function User(props: inferSSRProps<typeof getServerSideProps>) {
     </>
   );
 }
-
-export const getServerSideProps = async (context: GetServerSidePropsContext) => {
-  const username = (context.query.user as string).toLowerCase();
-
-  const user = await prisma.user.findUnique({
-    where: {
-      username,
-    },
+export const getStaticPaths: GetStaticPaths = async () => {
+  const allUsers = await prisma.user.findMany({
     select: {
-      id: true,
       username: true,
-      email: true,
-      name: true,
-      bio: true,
-      avatar: true,
-      theme: true,
-      plan: true,
     },
   });
-  if (!user) {
+  const usernames = allUsers.flatMap((u) => (u.username ? [u.username] : []));
+  return {
+    paths: usernames.map((user) => ({
+      params: { user },
+    })),
+
+    // https://nextjs.org/docs/basic-features/data-fetching#fallback-blocking
+    fallback: "blocking",
+  };
+};
+
+export const getStaticProps = async (context: GetStaticPropsContext<{ user: string }>) => {
```

Comment:
Prefetches the data on the page for each user and hydrates a cache that `react-query` uses on mount